core/interfaces/i_navigation_policy.ts

```diff
@@ -57,4 +56,13 @@ export interface INavigationPolicy<T> {
    * @returns True if this element should be included in keyboard navigation.
    */
   isNavigable(current: T): boolean;
+
+  /**
+   * Returns whether or not this navigation policy corresponds to the type of
+   * the given object.
+   *
+   * @param current An instance to check whether this policy applies to.
+   * @returns True if the given object is of a type handled by this policy.
```

Comment:
A nit but I don't think this actually returns a boolean, I think it returns `current` coerced to type `T` or `undefined` if coercion fails per the predicate.

I'm not 100% certain on this and I find the TypeScript documentation unclear, but I recall debugging guards like this before and being surprised that their return values weren't actually booleans.

Reply:
I agree the docs are unclear on this, but I think `boolean` is more useful to both folks reading and implementing than the perhaps technically correct value; ultimately the method needs to return a boolean (if you're implementing it) and you can rely on it being a truthy value if you call it.

Reply:
I used the language 'whether' in this case which I suppose is slightly more ambiguous (perhaps beneficially):

```typescript
/**
 * Determines whether the provided object fulfills the contract of
 * IFocusableNode.
 *
 * @param object The object to test.
 * @returns Whether the provided object can be used as an IFocusableNode.
 */
export function isFocusableNode(object: any | null): object is IFocusableNode {
  return (
    object &&
    'getFocusableElement' in object &&
    'getFocusableTree' in object &&
    'onNodeFocus' in object &&
    'onNodeBlur' in object &&
    'canBeFocused' in object
  );
}
```

Not sure that's actually better. Note I don't have strong opinions here, so if you prefer the explicit 'True' that seems reasonable to me.